File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/mappers/menus/ApiMenuContentMapper.java
package com.gigigo.orchextra.core.data.mappers.menus;

import com.gigigo.orchextra.core.data.dto.menus.ApiMenuContent;
import com.gigigo.orchextra.core.domain.entities.elements.Element;
import com.gigigo.orchextra.core.data.dto.elements.ApiElement;
import com.gigigo.orchextra.core.data.mappers.elements.ApiElementMapper;
import com.gigigo.orchextra.core.domain.entities.menus.MenuContent;
import com.gigigo.ggglib.mappers.ExternalClassToModelMapper;
import java.util.ArrayList;
import java.util.List;

public class ApiMenuContentMapper
    implements ExternalClassToModelMapper<ApiMenuContent, MenuContent> {

  private final ApiElementMapper apiMenuItemMapper;

  public ApiMenuContentMapper(ApiElementMapper apiMenuItemMapper) {
    this.apiMenuItemMapper = apiMenuItemMapper;
  }

  @Override public MenuContent externalClassToModel(ApiMenuContent data) {
    MenuContent model = new MenuContent();

    model.setId(data.getId());
    model.setSlug(data.getSlug());

    List<Element> menuItemList = new ArrayList<>();
    if (data.getElements() != null) {
      for (ApiElement apiMenuItem : data.getElements()) {
        menuItemList.add(apiMenuItemMapper.externalClassToModel(apiMenuItem));
      }
    }

    model.setElements(menuItemList);

    return model;
  }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/mappers/menus/ApiMenuContentMapper.kt
package com.gigigo.orchextra.core.data.mappers.menus

import android.util.Log
import com.gigigo.ggglib.mappers.ExternalClassToModelMapper
import com.gigigo.orchextra.core.data.api.dto.menus.ApiMenuContent
import com.gigigo.orchextra.core.data.mappers.elements.ApiElementMapper
import com.gigigo.orchextra.core.domain.entities.elements.Element
import com.gigigo.orchextra.core.domain.entities.menus.MenuContent
import java.util.ArrayList

class ApiMenuContentMapper(
    private val apiMenuItemMapper: ApiElementMapper) : ExternalClassToModelMapper<ApiMenuContent, MenuContent> {

  override fun externalClassToModel(data: ApiMenuContent): MenuContent {
    val time = System.currentTimeMillis()

    val model = MenuContent()

    model.slug = data.slug

    val menuItemList = ArrayList<Element>()

    data.elements?.let {
      for (apiMenuItem in data.elements) {
        apiMenuItemMapper.externalClassToModel(apiMenuItem)?.let {
          menuItemList.add(it)
        }
      }
    }

    model.elements = menuItemList

    Log.v("TT - ApiMenuContent", ((System.currentTimeMillis() - time) / 1000).toString() + "")

    return model
  }
}